### PR TITLE
Misc ci updates

### PR DESCRIPTION
**Pull request by naik-aakash**
## Changes

1. Update github actions versions
2. Add auto update test durations job and script

### CODE REVIEW

**Comment by naik-aakash** [comment]
Hi @JaGeo, before you merge this PR, can you please check if the GitHub actions have the right to create PRs on this repo as well ?

**Review by JaGeo — APPROVED** [review]

**Pull request merged**
